src/ApiService/ApiService/onefuzzlib/VmssOperations.cs

```diff
@@ -195,26 +198,40 @@ private Task<string> GetInstanceIdForVmId(Guid scaleset, Guid vmId)
         => _cache.GetOrCreateAsync(new InstanceIdKey(scaleset, vmId), async entry => {
```

Comment:
Saving the instance id makes the caching obsolete. We should only have one of them. I vote for keeping the caching for now and add the intance_id once we have validated the migration is complete.

Reply:
We just have 2 layers of caching.

Having "local" caching is good - makes things faster, and less requests to Azure storage required.

If my change is very risky - makes sense to revert. If change is low risk - then it has benefit

Reply:
I think if we don't have an immediate needs we should not add it now. However,  if we have to have 2 layers of caching let's not intermingle them in the same function it makes the logic complex.

This function should only be used for caching the azure request. I suggest saving the instance  to the database in the `GetInstanceId` method instead

Reply:
If we move it to GetInstanceId - then it defeats the point of storing it in the table at all. We want to store instance ids when we iterate through scaleset so we avoid Azure requests later on.

Moving to GetInstanceId will only cache one entry of the VM that was found.